manifests/social/slack.pp

```diff
@@ -0,0 +1,42 @@
+# slack.pp
+# Install Slack for OS X, Ubuntu, or Windows
+# https://www.slack.com, https://packagecloud.io/slacktechnologies
+#
+
+class software::social::slack (
+  $ensure = $software::params::software_ensure,
+) inherits software::params {
+
+  validate_string($ensure)
+
+  case $::operatingsystem {
+    'Ubuntu': {
+      $apt_source_ensure = $ensure ? {
+        'installed' => present,
+        'latest'    => present,
+        default     => $ensure,
+      }
+
+      include '::apt'
+      apt::source { 'slack':
+        ensure   => $apt_source_ensure,
+        comment  => 'Slack APT sources for any Debian-based distribution, including Ubuntu',
+        location => 'https://packagecloud.io/slacktechnologies/slack/debian',
+        release  => 'jessie',
```

Comment:
Is this always jessie, even when on Ubuntu? Isn't that a release name for Debian?

Reply:
Yes, it always is. See https://packagecloud.io/slacktechnologies/slack

Reply:
Another thing, when you install Slack "the usual way" then the APT sources contain just what the `location` in the Puppet code corresponds to. And with Puppet I've done the test drive on my Ubuntu Zesty box:

```bash
$ cat /etc/apt/sources.list.d/slack.list 
# This file is managed by Puppet. DO NOT EDIT.
# Slack APT sources, from https://packagecloud.io/slacktechnologies/slack (Puppet module broken)
deb https://packagecloud.io/slacktechnologies/slack/debian jessie main
```
```bash
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty
```

Reply:
OK, I though this was the case from some of your other comments, but wanted to verify.